internal/helper/helper.go

```diff
@@ -201,64 +201,86 @@ func CheckEmptyFields(Interface interface{}, result *gojsonschema.Result, schema
-						return err
-					}
-				}
+			err := checkEmptyFieldsInSlice(field, result, schema)
+			if err != nil {
+				return err
 			}
 		} else if field.Type().Kind() == reflect.Ptr {
 			// otherwise if it's just a pointer to a nested struct
 			// search it for empty required fields
-			if field.Elem().Kind() == reflect.Struct {
-				err := CheckEmptyFields(field.Interface(), result, schema)
-				if err != nil {
-					return err
-				}
+			err := checkEmptyFieldsInPtr(field, result, schema)
+			if err != nil {
+				return err
 			}
 		} else {
-
-			// check if the field is required and if it is present in the YAML file
-			required := false
-			tags := elem.Type().Field(i).Tag
-			jsonTag, hasJSON := tags.Lookup("json")
-			if hasJSON {
-				if !strings.Contains(jsonTag, "omitempty") {
-					required = true
-				}
+			required, tags := isRequired(elem, i, schema)
+			if !required {
+				continue
 			}
-			// also check for required values in the jsonschema
-			for _, requiredField := range schema.Required {
-				if elem.Type().Field(i).Name == requiredField {
-					required = true
-				}
+			// this is a required field, check for zero values
+			if !reflect.Indirect(field).IsZero() {
+				continue
 			}
-			if required {
-				// this is a required field, check for zero values
-				if reflect.Indirect(field).IsZero() {
-					jsonContext := gojsonschema.NewJsonContext("image_definition", nil)
-					errDetail := gojsonschema.ErrorDetails{
-						"property": tags.Get("yaml"),
-						"parent":   elem.Type().Name(),
-					}
-					result.AddError(
-						newMissingFieldError(
-							gojsonschema.NewJsonContext("missing_field", jsonContext),
-							52,
-							errDetail,
-						),
-						errDetail,
-					)
-				}
+			jsonContext := gojsonschema.NewJsonContext("image_definition", nil)
+			errDetail := gojsonschema.ErrorDetails{
+				"property": tags.Get("yaml"),
+				"parent":   elem.Type().Name(),
+			}
+			result.AddError(
+				newMissingFieldError(
+					gojsonschema.NewJsonContext("missing_field", jsonContext),
+					52,
+					errDetail,
+				),
+				errDetail,
+			)
+		}
+	}
+	return nil
+}
+
+func checkEmptyFieldsInSlice(field reflect.Value, result *gojsonschema.Result, schema *jsonschema.Schema) error {
+	for i := 0; i < field.Cap(); i++ {
+		sliceElem := field.Index(i)
+		if sliceElem.Kind() == reflect.Ptr && sliceElem.Elem().Kind() == reflect.Struct {
+			err := CheckEmptyFields(sliceElem.Interface(), result, schema)
+			if err != nil {
+				return err
 			}
 		}
 	}
 	return nil
 }
 
+func checkEmptyFieldsInPtr(field reflect.Value, result *gojsonschema.Result, schema *jsonschema.Schema) error {
+	if field.Elem().Kind() == reflect.Struct {
+		err := CheckEmptyFields(field.Interface(), result, schema)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// isRequired checks if the field is required and if it is present in the YAML file
+func isRequired(elem reflect.Value, i int, schema *jsonschema.Schema) (bool, reflect.StructTag) {
```

Comment:
This method is doing more that its name suggests. It seems like the second part of it should be another method. And I'd not expect a method called `isRequired` to return a bool AND something else, in this case tags. Specially because `tags` is not related to the fact the field is required or not, it will be returned regardless.